src/devices/tuya.ts

```diff
@@ -8712,6 +8701,420 @@ const definitions: Definition[] = [
+                attribute: 'value',
+                actionLookup: {'button': 0},
+                buttonLookup: {
+                    '1_up': 4, '1_down': 1,
+                    '2_up': 5, '2_down': 2,
+                    '3_up': 6, '3_down': 3,
+                },
+            }),
+        ],
+    },
+    {
+        fingerprint: tuya.fingerprint('TS0001', ['_TZ3000_gtdswg8k']),
+        model: 'QARZDC1LR',
+        vendor: 'QA',
+        description: 'QA 1CH Long Range',
+        extend: [
+            tuya.modernExtend.tuyaMagicPacket(),
+            deviceEndpoints({endpoints: {'l1': 1}}),
+            tuya.modernExtend.tuyaOnOff({endpoints: ['l1'], powerOnBehavior2: true, switchType: true}),
+        ],
+    },
+    {
+        fingerprint: tuya.fingerprint('TS0002', ['_TZ3000_rfjctviq', '_TZ3210_pdnwpnz5', '_TZ3210_a2erlvb8']),
+        model: 'QARZ2LR',
+        vendor: 'QA',
+        description: 'QA 2CH Long Range',
+        extend: [
+            tuya.modernExtend.tuyaMagicPacket(),
+            deviceEndpoints({endpoints: {'l1': 1, 'l2': 2}}),
+            tuya.modernExtend.tuyaOnOff({endpoints: ['l1', 'l2'], powerOnBehavior2: true, switchType: true}),
+        ],
+    },
+    {
+        fingerprint: tuya.fingerprint('TS0003', ['_TZ3000_zeuulson']),
+        model: 'QARZ3LR',
+        vendor: 'QA',
+        description: 'QA 3CH Long Range',
+        extend: [
+            tuya.modernExtend.tuyaMagicPacket(),
+            deviceEndpoints({endpoints: {'l1': 1, 'l2': 2, 'l3': 3}}),
+            tuya.modernExtend.tuyaOnOff({endpoints: ['l1', 'l2', 'l3'], powerOnBehavior2: true, switchType: true}),
+        ],
+    },
+    {
+        fingerprint: tuya.fingerprint('TS0004', ['_TZ3000_wwtnshol']),
+        model: 'QARZ4LR',
+        vendor: 'QA',
+        description: 'QA 4CH Long Range',
+        extend: [
+            tuya.modernExtend.tuyaMagicPacket(),
+            deviceEndpoints({endpoints: {'l1': 1, 'l2': 2, 'l3': 3, 'l4': 4}}),
+            tuya.modernExtend.tuyaOnOff({endpoints: ['l1', 'l2', 'l3', 'l4'], powerOnBehavior2: true, switchType: true}),
+        ],
+    },
+    {
+        fingerprint: tuya.fingerprint('TS1201', ['_TZ3290_jxvzqatwgsaqzx1u']),
+        model: 'QAIRZPRO',
+        vendor: 'QA',
+        description: 'QA IR Hub',
+        whiteLabel: [{vendor: 'QA', model: 'QAIRZPRO'}],
+        fromZigbee: [
+            zosung.fzZosung.zosung_send_ir_code_00, zosung.fzZosung.zosung_send_ir_code_01, zosung.fzZosung.zosung_send_ir_code_02,
+            zosung.fzZosung.zosung_send_ir_code_03, zosung.fzZosung.zosung_send_ir_code_04, zosung.fzZosung.zosung_send_ir_code_05,
+        ],
+        toZigbee: [zosung.tzZosung.zosung_ir_code_to_send, zosung.tzZosung.zosung_learn_ir_code],
+        exposes: [zosung.presetsZosung.learn_ir_code(), zosung.presetsZosung.learned_ir_code(), zosung.presetsZosung.ir_code_to_send()],
+    },
+    {
+        fingerprint: tuya.fingerprint('TS0001', [
```

Comment:
Add as a whiteLabel to `TS0001_switch_1_gang`

Reply:
We use different exposes for this device:
extend: [
            tuya.modernExtend.tuyaMagicPacket(),
            deviceEndpoints({endpoints: {'l1': 1}}),
            tuya.modernExtend.tuyaOnOff({endpoints: ['l1'], powerOnBehavior2: true, switchType: true}),
        ],